From b011a799339fb61c7f2b8b9c9c169c168d1c98c7 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinh.nguyen@intel.com>
Date: Thu, 26 Aug 2021 09:26:34 -0500
Subject: [PATCH 37/42] clk: socfpga: agilex: fix the parents of the
 psi_ref_clk

commit eaa2e18bedd1336e1bb7496d2c0dab25a1dc097b from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

[upstream commit 9d563236cca43fc4fe190b3be173444bd48e2a3b]

The psi_ref_clk comes from the C2 node of the main_pll and periph_pll,
not the C3.

Fixes: 80c6b7a0894f ("clk: socfpga: agilex: add clock driver for the Agilex platform")
Cc: stable@vger.kernel.org
Signed-off-by: Kris Chaplin <kris.chaplin@intel.com>
Signed-off-by: Dinh Nguyen <dinguyen@kernel.org>
Link: https://lore.kernel.org/r/20210713144621.605140-1-dinguyen@kernel.org
Signed-off-by: Stephen Boyd <sboyd@kernel.org>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/clk/socfpga/clk-agilex.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/socfpga/clk-agilex.c b/drivers/clk/socfpga/clk-agilex.c
index 07ccb569d7c0..ab86e98fc628 100644
--- a/drivers/clk/socfpga/clk-agilex.c
+++ b/drivers/clk/socfpga/clk-agilex.c
@@ -39,7 +39,7 @@ static const char * const emac_ptp_free_mux[] = {"main_pll_c3", "peri_pll_c3",
 static const char * const gpio_db_free_mux[] = {"main_pll_c3", "peri_pll_c3",
 					      "osc1", "cb-intosc-hs-div2-clk",
 					      "f2s-free-clk"};
-static const char * const psi_ref_free_mux[] = {"main_pll_c3", "peri_pll_c3",
+static const char * const psi_ref_free_mux[] = {"main_pll_c2", "peri_pll_c2",
 					      "osc1", "cb-intosc-hs-div2-clk",
 					      "f2s-free-clk"};
 static const char * const sdmmc_free_mux[] = {"main_pll_c3", "peri_pll_c3",
-- 
2.31.1

